boards/shields/openthread_rcp_spi/openthread_rcp_spi.overlay

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2025 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Overlay to enable support for OpenThread's RCP over SPI communication
+ */
+
+/ {
+	chosen {
+		zephyr,hdlc-rcp-if = &hdlc_rcp_if;
+	};
+};
+
+&arduino_spi {
+	status = "okay";
+
+	hdlc_rcp_if: hdlc_rcp_if@0 {
+		compatible = "spi,hdlc-rcp-if";
```

Comment:
I'm not a fan myself of this compatible, as it does prints out:

```
node '/soc/spi@40003800/hdlc_rcp_if@0' compatible 'spi,hdlc-rcp-if' has unknown vendor prefix 'spi'
```

But this is consistent with the other `uart,hdlc-rcp-if` compatible 🤷‍♂️